src/control/cmd/dmg/storage_query.go

```diff
@@ -81,78 +116,69 @@ func (cmd *nvmeHealthQueryCmd) Execute(args []string) error {
 	return resp.Errors()
 }
 
-// bsHealthQueryCmd is the struct representing the "storage query bio" subcommand
-//
-// Command is issued to the management service access point.
-type bsHealthQueryCmd struct {
-	logCmd
-	connectedCmd
-	Devuuid string `short:"u" long:"devuuid" description:"Device/Blobstore UUID to query"`
-	Tgtid   string `short:"t" long:"tgtid" description:"VOS target ID to query"`
+type devHealthQueryCmd struct {
+	smdQueryCmd
+	UUID string `short:"u" long:"uuid" required:"1" description:"Device UUID"`
 }
 
-// Execute is run when bsHealthQueryCmd activates.
-// Query the BIO health and error stats of the given device.
-func (b *bsHealthQueryCmd) Execute(args []string) error {
-	if b.Devuuid != "" && b.Tgtid != "" {
-		return errors.New("either device UUID OR target ID need to be specified not both")
-	} else if b.Devuuid == "" && b.Tgtid == "" {
-		return errors.New("device UUID or target ID is required")
+func (cmd *devHealthQueryCmd) Execute(_ []string) error {
+	ctx := context.Background()
+	req := &control.SmdQueryReq{
+		OmitPools:        true,
+		IncludeBioHealth: true,
+		Rank:             system.NilRank,
+		UUID:             cmd.UUID,
 	}
-
-	req := &mgmtpb.BioHealthReq{DevUuid: b.Devuuid, TgtId: b.Tgtid}
-
-	b.log.Infof("Blobstore Health Data:\n%s\n", b.conns.BioHealthQuery(req))
-
-	return nil
+	return cmd.makeRequest(ctx, req)
 }
 
-// smdQueryCmd is the struct representing the "storage query smd" subcommand
-//
-// Command is issued to the management service access point.
-type smdQueryCmd struct {
-	logCmd
-	connectedCmd
-	Devices bool `short:"d" long:"devices" description:"List all devices/blobstores stored in per-server metadata table."`
-	Pools   bool `short:"p" long:"pools" description:"List all VOS pool targets stored in per-server metadata table."`
+type tgtHealthQueryCmd struct {
+	smdQueryCmd
+	Rank  uint32 `short:"r" long:"rank" required:"1" description:"Server rank hosting target"`
+	TgtId uint32 `short:"t" long:"tgtid" required:"1" description:"VOS target ID to query"`
 }
 
-// Execute is run when ListSmdDevCmd activates
-// Query per-server metadata device table for all connected servers
-func (s *smdQueryCmd) Execute(args []string) error {
-	// default is to print both pools and devices if not specified
-	if !s.Pools && !s.Devices {
-		s.Pools = true
-		s.Devices = true
+func (cmd *tgtHealthQueryCmd) Execute(_ []string) error {
+	ctx := context.Background()
+	req := &control.SmdQueryReq{
+		OmitPools:        true,
+		IncludeBioHealth: true,
+		Rank:             system.Rank(cmd.Rank),
+		Target:           strconv.Itoa(int(cmd.TgtId)),
 	}
+	return cmd.makeRequest(ctx, req)
+}
 
-	if s.Devices {
-		req_dev := &mgmtpb.SmdDevReq{}
-		s.log.Infof("SMD Device List:\n%s\n", s.conns.SmdListDevs(req_dev))
-	}
+type listDevicesQueryCmd struct {
+	smdQueryCmd
+	rankCmd
+	Health bool   `short:"b" long:"health" description:"Include BIO health in results"`
```

Comment:
Maybe avoid "BIO" since that will not make sense to an admin, instead just "Include device health"?

Reply:
Done, thanks.